packages/admin/cms-admin/src/pages/pageTree/useCopyPastePages/sendPages.tsx

```diff
@@ -351,7 +355,16 @@ export async function sendPages(
                 }
             }
 
-            // 3b. Create new document and attach it to new page tree node
+            // 3b. Replace unhandled dependencies with undefined (when copying to another scope)
+            if (sourcePage.document && !isEqual(rootContentScope, targetContentScope)) {
+                const unhandledDependencies = unhandledDependenciesFromDocument(documentType, sourcePage.document, {
+                    existingReplacements: dependencyReplacements,
+                });
+                const replacementsForUnhandledDependencies = createUndefinedReplacementsForDependencies(unhandledDependencies);
+                dependencyReplacements.push(...replacementsForUnhandledDependencies);
+            }
```

Comment:
I don't handle the case of copying to another environment. Maybe we don't have to handle it since the target `id` doesn't exist there anyway, so no info can leak between scopes.

The internal link is flexible enough to handle non-existing ids. In theory, all blocks should be that flexible. Though I'm not sure if that's actually the case.

Reply:
You could try if this is a problem with the DAM blocks, for instance `PixelImageBlock`

Reply:
@johnnyomair I just tested this. PixelImageBlock and SvgImageBlock are both flexible enough to handle non-existing IDs.

Usually, files are handled by downloading them from the source environment. So having non-existent file IDs should be an exception